### PR TITLE
Add UI to show the calculated guide star

### PR DESCRIPTION
**Pull request by cquiroz**
This was fun

### CODE REVIEW

File: explore/src/main/scala/explore/targeteditor/AladinCell.scala [hunk 1]
```diff
@@ -73,16 +98,34 @@ object AladinCell extends ModelOptics {
       // avoids us needing a ref to a Fn component
       .useStateView(false)
       // Listen on web worker for messages with catalog candidates
-      .useStreamOnMountBy((props, _, _, _) =>
-        props.ctx.worker.stream
-          .map(decodeFromTransferable[CatalogResults])
-          .unNone
-          .map(_.candidates)
+      .useStreamBy((props, _, _, _) => (props.target.get, props.obsConf.map(_.obsInstant)))(
```
Comment (toddburnside) [inline explore/src/main/scala/explore/targeteditor/AladinCell.scala:101]: Just curious. What prompted the change from `useStreamOnMountBy` to `useStreamBy`.

**Review by toddburnside — APPROVED** [review]
Wow, that took a lot of changes! LGTM.

File: explore/src/main/scala/explore/targeteditor/AgsOverlay.scala [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+final case class AgsOverlay(
+  selectedGSIndex:     ReuseView[Int],
+  guideStarCandidates: List[(GuideStarCandidate, AgsAnalysis)]
+) extends ReactFnProps[AgsOverlay](AgsOverlay.component)
+
+object AgsOverlay {
+  type Props = AgsOverlay
+
+  // This is used for fov thus needs to be fairly smal
+  implicit val propsReuse: Reusability[Props] = Reusability.derive
```
Comment (rpiaggio) [inline explore/src/main/scala/explore/targeteditor/AgsOverlay.scala:35]: Shall we stick to not using `Reusability` unless necessary?

File: explore/src/main/scala/explore/targeteditor/AladinCell.scala [hunk 3]
```diff
@@ -73,187 +98,240 @@ object AladinCell extends ModelOptics {
...
+              }.getOrElse(Nil)
+            case _                                            => Nil
+          }
+      }
       // open settings menu
       .useState(false)
-      .render { (props, mouseCoords, options, center, gsc, openSettings) =>
-        implicit val ctx = props.ctx
+      // Selected GS index
+      .useStateViewWithReuse(0)
```
Comment (rpiaggio) [inline explore/src/main/scala/explore/targeteditor/AladinCell.scala:170]: No need for `withReuse`

**Review by rpiaggio — APPROVED** [review]
LGTM, looks awesome

File: explore/src/main/scala/explore/targeteditor/AgsOverlay.scala (hunk 2 above)
Comment (cquiroz) [reply explore/src/main/scala/explore/targeteditor/AgsOverlay.scala:35]: Yes, though is some cases they are needed for the hooks

File: explore/src/main/scala/explore/targeteditor/AladinCell.scala [hunk 4]
```diff
@@ -73,187 +98,240 @@ object AladinCell extends ModelOptics {
...
+                val basePos = AgsPosition(pa, Offset.Zero)
+                Ags
+                  .agsAnalysis[IO](bestConstraintSet,
+                                   wavelength,
+                                   tracking.baseCoordinates,
+                                   basePos,
+                                   params,
+                                   candidates
+                  )
+                  .sorted
```
[the remark this answers is not in the document]
Comment (cquiroz) [reply explore/src/main/scala/explore/targeteditor/AladinCell.scala:162]: You're right

**Pull request merged**